File: JMS.DAL/Models/JourneyUpdate.cs
using JMS.DAL.Common.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace JMS.DAL.Models
{
    public partial class JourneyUpdate
    {
        public JourneyUpdate()
        {
            AssessmentResult = new HashSet<AssessmentResult>();
        }

        public int Id { get; set; }
        public DateTime? Date { get; set; }
        public int? JourneyId { get; set; }
        public JourneyStatus JourneyStatus { get; set; }
        public string VehicleNo { get; set; }
        public string DriverId { get; set; }
        public bool IsJourneyCheckpoint { get; set; }
        public int? CheckpointId { get; set; }
        public Level RiskLevel { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public bool IsDriverStatus { get; set; }
        public bool IsAlert { get; set; }
        public string StatusMessage { get; set; }
        public Guid? UserId { get; set; }

        public virtual Checkpoint Checkpoint { get; set; }
        public virtual Journey Journey { get; set; }
        public virtual ICollection<AssessmentResult> AssessmentResult { get; set; }
        public virtual User User { get; set; }
    }
}

File: JMS.WEB/GMSSolution/src/app/shared/Entities/User/journey-update.ts
import { Level } from './level.enum';
import { JourneyStatus } from './journey-status.enum';
import { Checkpoint } from './checkpoint';
import { Journey } from './journey';
import { AssessmentResult } from './assessment-result';
import { User } from './user';

export class JourneyUpdate {
  Id: Number;
  Date: Date;
  JourneyId: Number;
  JourneyStatus: JourneyStatus;
  VehicleNo: string;
  DriverId: string;
  IsJourneyCheckpoint: boolean;
  CheckpointId: Number;
  RiskLevel: Level;
  Latitude: Number;
  Longitude: Number;
  IsDriverStatus:boolean;
  IsAlert:boolean;
  StatusMessage:string;
  UserId:string;
  Checkpoint:Checkpoint;
  Journey:Journey;
  AssessmentResult:AssessmentResult[];
  User:User;
}
